libsolidity/formal/SolverInterface.h

```diff
@@ -199,8 +199,10 @@ class SolverInterface
 	virtual void push() = 0;
 	virtual void pop() = 0;
 
+	virtual void declareFunction(std::string _name, Sort _domain, Sort _codomain) = 0;
 	virtual Expression newFunction(std::string _name, Sort _domain, Sort _codomain)
 	{
+		declareFunction(_name, _domain, _codomain);
```

Comment:
Basically previously the child class was calling the parent, now it is the other opposite.

Reply:
Yes. The idea now is that the subclasses only implement the declaration, because that's solver dependent. Returning the actual expression is common and can be done in the parent. Perhaps the `new...` functions in the parent shouldn't be virtual.